tmt/base.py

```diff
@@ -1587,13 +1588,30 @@ def import_plan(self) -> Optional['Plan']:
                 else:
                     tmt.utils.git_clone(plan_id.url, destination, self)
                 if plan_id.ref:
-                    self.run(Command('git', 'checkout', plan_id.ref), cwd=destination)
+                    # attempt to evaluate dynamic reference
+                    try:
+                        ref = resolve_dynamic_ref(
+                            workdir=destination,
+                            ref=plan_id.ref,
+                            plan=self,
+                            logger=self._logger
+                            )
+                    except tmt.utils.FileError as error:
+                        raise tmt.utils.DiscoverError(str(error))
```

Comment:
I'd recommend using `from` here, that way upper levels can provide more context while preserving the original lower level exception which might carry some interesting info. For example, the following would be nice to see if there's an error:

```python
except tmt.utils.FileError as error:
    raise tmt.utils.DiscoverError(f"Failed to resolve dynamic ref of '{plan_id.ref}'.") from error
```

Reply:
Fixed